lib/portlayer/storage/vsphere/volume.go

```diff
@@ -87,6 +87,28 @@ func (v *VolumeStore) AddStore(ctx context.Context, ds *datastore.Helper, storeN
 	return u, nil
 }
 
+func (v *VolumeStore) VolumeStoresList(ctx context.Context) (map[string]url.URL, error) {
+	m := make(map[string]url.URL)
+	for u, ds := range v.ds {
+
+		// Get the ds:// URL given the datastore url ("[datastore] /path")
+		dsurl, err := datastore.ToURL(ds.RootURL)
```

Comment:
In order to avoid copies of "ds" you could replace `ds.RootURL` with v[u].RootURL